src/tools/netkey-tool.c

```diff
@@ -542,17 +542,20 @@ int main(
 		if(!strcmp("tcos", ctx->card_drivers[i]->short_name)) break;
 	if(!ctx->card_drivers[i]){
 		fprintf(stderr,"Context does not support TCOS-cards\n");
+		sc_release_context(ctx);
```

Comment:
You should initialize the `ctx` to NULL on line 440 to make sure we are not accessing uninitialized memory if something goes wrong. It is not a must, but it would be great as we are already touching the code.